charms/kubernetes_snaps.py

```diff
@@ -559,32 +571,67 @@ def get_sandbox_image(registry) -> str:
     return f"{registry}/pause:3.9"
 
 
+def get_snap_version(name: str) -> str:
+    """
+    Get the version of an installed snap package.
+
+    Args:
+    name (str): The name of the snap package.
+
+    Returns:
+    str or None: The version of the snap package if available, otherwise None.
+    """
+    cmd = ["snap", "list", name]
+    result = check_output(cmd)
+    output = result.decode().strip()
+    match = re.search(r"\b\d+(?:\.\d+)*\b", output)
+
+    if match:
+        return match.group()
+    else:
+        log.info(f"Package '{name}' not found or no version available.")
+    return None
+
+
 def host_is_container():
     return call(["systemd-detect-virt", "--container"]) == 0
 
 
 @status.on_error(BlockedStatus("Failed to install Kubernetes snaps"))
-def install(channel, control_plane=False):
+def install(channel, control_plane=False, upgrade=False):
     """Install or refresh Kubernetes snaps. This includes the basic snaps to
-    talk to Kubernetes and run a Kubernetes nodes.
-
-    If control_plane=True, then also install the Kubernetes control plane snaps.
+    talk to Kubernetes and run a Kubernetes node.
+
+    Args:
+        - channel (str): The snap channel to install from.
+        - control_plane (bool, optional): If True, installs the Kubernetes control
+        plane snaps. Defaults to False.
+        - upgrade (bool, optional): If True, allows upgrading of snaps. Defaults to
+        False.
     """
 
+    if any(is_upgrade(snap, channel) for snap in BASIC_SNAPS) and not upgrade:
+        status.add(
+            BlockedStatus("Snap channel version has changed. An upgrade is required.")
+        )
```

Comment:
@mateoflorido do you see any trouble changing this text to match the previous message in reactive:

```
Needs manual upgrade, run the upgrade action
```